core/domain/fs_domain.py

```diff
@@ -108,96 +107,105 @@ def version(self):
         return self._version
 
 
-class ExplorationFileSystem(object):
-    """A datastore-backed read-write file system for a single exploration.
+class EntityFileSystem(object):
+    """A datastore-backed read-write file system for a single entity.
 
-    The conceptual intention is for each exploration to have its own asset
-    folder. An asset has no meaning outside its exploration, so the assets in
-    these asset folders should therefore not be edited directly. They should
-    only be modified as side-effects of some other operation (such as adding an
-    image to an exploration).
-
-    The content of an exploration should include a reference to the asset
-    together with the version number of the asset. This allows the
-    exploration to refer to asset versions.
+    The conceptual intention is for each entity type to have its own parent
+    folder. In this, each individual object will have its own folder with the
```

Comment:
By "object" do you mean "entity"? Let's use consistent language.

Reply:
Hmm, not really. By entity, we mean 'exploration' or 'question' in general, right?  This here means an instance of that entiity, like a particular exploration. So, should I change it to 'individual instance of an entity'?

Reply:
Done